From 6346fbef248ddf1529caced6b30672fb96ef6d01 Mon Sep 17 00:00:00 2001
Message-Id: <6346fbef248ddf1529caced6b30672fb96ef6d01.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: "A.Sunil Kamath" <sunil.kamath@intel.com>
Date: Wed, 21 Aug 2013 16:48:57 +0530
Subject: [PATCH 008/312] MUST_UPSTREAM [VPG]: drm/i915: HDCP: Notify HPD to
 userspace

This change is expected as part HDCPD compliance.
HDCPD expects an uevent from display driver with every HPD.
This change will ensure to send an uevent from kernel space to
userspace, every time we get a hot plug event on HDMI.
The HDCP daemon polls for this event and takes necessary
action from userspace. Compliance test passes succesfully
with this change.

Change-Id: I07fe025d86652f05f97ce185e544215d653147eb
Signed-off-by: Shashank Sharma <shashank.sharma@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
(cherry picked from commit 1a029feb2977f7bb178634f9f4988066f291ddc8)

Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index 69a7960..a2f79b2 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -1108,6 +1108,7 @@ static void i915_hotplug_work_func(struct work_struct *work)
 	bool hpd_disabled = false;
 	bool changed = false;
 	u32 hpd_event_bits;
+	char *envp[] = {"hdcp_hpd", NULL};
 
 	mutex_lock(&mode_config->mutex);
 	DRM_DEBUG_KMS("running encoder hotplug functions\n");
@@ -1160,6 +1161,11 @@ static void i915_hotplug_work_func(struct work_struct *work)
 
 	if (changed)
 		drm_kms_helper_hotplug_event(dev);
+
+	/* HDCPD needs a uevent, every time when there is a hotplug */
+	kobject_uevent_env(&dev_priv->dev->primary->kdev->kobj,
+		KOBJ_CHANGE, envp);
+
 }
 
 static void intel_hpd_irq_uninstall(struct drm_i915_private *dev_priv)
-- 
1.7.9.5

